storage/integration_test.go

```diff
@@ -6305,6 +6305,48 @@ func TestIntegration_ReaderCancel(t *testing.T) {
 	})
 }
 
+func TestIntegration_ListBuckets(t *testing.T) {
+	ctx := skipExtraReadAPIs(context.Background(), "no reads in test")
+	multiTransportTest(ctx, t, func(t *testing.T, ctx context.Context, _ string, prefix string, client *Client) {
+		h := testHelper{t}
+		projectID := testutil.ProjID()
+
+		newBucketName := prefix + uidSpace.New()
+		bkt := client.Bucket(newBucketName)
+
+		h.mustCreate(bkt, projectID, nil)
+		t.Cleanup(func() { h.mustDeleteBucket(bkt) })
+
+		for _, partialSuccess := range []bool{true, false} {
+			t.Run(fmt.Sprintf("partialSuccess=%v", partialSuccess), func(t *testing.T) {
+				it := client.Buckets(ctx, projectID)
```

Comment:
This might take a long time to run if there are a lot of buckets in the project. Maybe use Prefix to filter the response to just the one you want?